PhpcrMigration/Infrastructure/Repository/EntityRepository.php

```diff
@@ -193,4 +209,48 @@ public function addOrUpdateChildNode(array $data, string $tableName, array $type
 
         $this->connection->insert($tableName, $data, $types);
     }
+
+    private function getNextIdValue(string $tableName): ?int
+    {
+        $result = null;
+
+        $platform = $this->connection->getDatabasePlatform();
+        $sequences = $this->getSequences();
+        foreach ($sequences as $sequence) {
+            $sequenceName = $sequence->getName();
+            if (\str_contains($sequenceName, $tableName)) {
+                /** @var int|null|false $result */
+                $result = $this->connection->fetchOne(
+                    $platform->getSequenceNextValSQL($sequenceName)
+                );
+
+                if (false === $result || null === $result) {
+                    throw new \RuntimeException('Failed to get next ID value from sequence' . $sequenceName);
+                }
+
+                break;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @return Sequence[]
+     */
+    private function getSequences(): array
+    {
+        if (isset($this->sequences)) {
+            return $this->sequences;
+        }
+
+        $sequences = [];
+        try {
+            $sequences = $this->connection->createSchemaManager()->listSequences();
```

Comment:
Does this trigger a database query? if not I would move it to the `__construct` method, else from strictness we would else require to reset `$this->sequences = []` on kernel reset.

Reply:
yes this does trigger one